docs/answers.md

```diff
@@ -5,6 +5,39 @@ title: Answers
 
 ---
 
+## Callbacks
+
+### Side Effects with **forEach**
+
+```{js}
+const doubleInPlace = (array) => {
+  array.forEach((value, location, container) => {
+    container[location] = value * 2;
+  })
+}
+```
+
+### Annotating Data
+
+```{js}
+const reducer = (accumulator, currentValue, index) => {
+    if (!accumulator) accumulator = []
+    if (currentValue.sex && currentValue.sex === 'F'){
+        accumulator.push({
+            'seq': index + 1,
+            'year': currentValue.date.split('-')[0],
+            'sex': currentValue.sex,
+            'species': currentValue.species
+        })
+    }
+    return accumulator
+}
+
+const annotateData = (array) => array.reduce(reducer, [])
+
+const newData = annotateData(data)
```

Comment:
This works, but using `.map` and `.filter` may be a more elegant solution? 

```
const cleanData = (d) => {
    const out = d.map((item, idx) => {
        return {
            seq: idx + 1,
            year: item.date.slice(0, 4),
            sex: item.sex,
            species: item.species
        }
    })
    .filter(x => x.sex === 'F')
    return out
}
```

Reply:
@cRAN-cg - this is more a suggestion than an issue that needs to be resolved, let me know what you think.

Reply:
Yes, I totally agree with you. I had a misconception of using Array.reducefor this solution as it was mentioned

    You will probably want to use Array.reduce to generate the sequence numbers.

We can include in both the solutions, explicitly indicating that the .map , .filter way is a more glorious way to do so.
What do you think about it @MayaGans

Reply:
Oh good catch - not a misconception at all, you're totally right. I'm going to merge as I think it'd be a little confusing to the reader if we present multiple explanations. Thanks!

Reply:
thanks :))